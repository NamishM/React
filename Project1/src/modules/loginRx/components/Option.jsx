import PropTypes from 'prop-types';
import React from 'react';

const Option = ({ name, id }) => (
  <option value={id}>{name}</option>
);

Option.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
};

export default Option;
