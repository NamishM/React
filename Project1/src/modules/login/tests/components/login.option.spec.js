/* eslint-disable react/jsx-filename-extension */
// REASON: Need to determine if *.jsx will be recodnized by
// mocha / enzyme configuration.

import React from 'react';
// import { shallow } from 'enzyme';
import Option from '../../components/Option';
import renderer from 'react-test-renderer';


const describe = global.describe;
const it = global.test;

const expect = global.expect;

describe('login -> components -> <Option />', () => {
  it('renders an <option tag with the id as the val and name as the textnode', () => {
    const component = renderer.create(
      <Option name="testName" id={1} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
