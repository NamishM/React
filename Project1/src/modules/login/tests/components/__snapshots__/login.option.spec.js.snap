// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login -> components -> <Option /> renders an <option tag with the id as the val and name as the textnode 1`] = `
<option
  value={1}
>
  testName
</option>
`;
